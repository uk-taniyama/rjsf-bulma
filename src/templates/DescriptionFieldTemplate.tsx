import type { FC } from "react";

import clsx from "clsx";

import { isSmallClass } from "../ui";

import type { DescriptionFieldProps } from "@rjsf/utils";

const DescriptionFieldTemplate: FC<DescriptionFieldProps> = ({
  schema: { type },
  description,
  registry,
}) => {
  if (!description) {
    return null;
  }

  return (
    <div
      className={clsx(
        type === "object" ? "description-field" : "help",
        isSmallClass(registry.formContext)
      )}
    >
      {description}
    </div>
  );
};

export default DescriptionFieldTemplate;
