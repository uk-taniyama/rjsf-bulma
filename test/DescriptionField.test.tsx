import React from "react";

import renderer from "react-test-renderer";

import DescriptionFieldTemplate from "../src/templates/DescriptionFieldTemplate";

import { mockRegistry } from "./helpers/createMocks";

const registry = mockRegistry();

describe("DescriptionField", () => {
  test("should return null when no description as a props is passed", () => {
    const tree = renderer
      .create(
        <DescriptionFieldTemplate
          id="testid"
          description=""
          registry={registry}
        />
      )
      .toJSON();
    expect(tree).toBe(null);
  });

  test("should return h2 element when description is being passed as props", () => {
    const tree = renderer
      .create(
        <DescriptionFieldTemplate
          id="testid"
          description="SOME THING"
          registry={registry}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
