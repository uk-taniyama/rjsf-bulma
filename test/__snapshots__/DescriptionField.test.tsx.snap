// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DescriptionField should return element when description is being passed as props 1`] = `
<div
  className="help"
>
  SOME THING
</div>
`;

exports[`DescriptionField should return element when description is being passed as props for object 1`] = `
<div
  className="description-field"
>
  SOME THING
</div>
`;
